### PR TITLE
Fix Rails Turbo support

### PR DESCRIPTION
**Pull request by WozniakMac**
Fixes issues:
https://github.com/themesberg/flowbite/issues/697
https://github.com/themesberg/flowbite/issues/698

BRAKING CHANGES:

`data-modal-toggle` will not work without `data-modal-target` anymore.

### CODE REVIEW

**Comment by zoltanszogyenyi** [comment]
Hey @WozniakMac,

Thanks for the PR! Couple of problems from my side:

- why did we remove the `data-modal-toggle` attribute?
- so what I see is that you've created another local instance manager: my issue here is that we should keep all instances, as much as possible under the main Instance Manager from `instances.ts` as possible
- what are the implications of the PR for the main vanilla version and other frameworks? This is in fact one of the trickiest things about the Flowbite JS

Cheers,
Zoltan

**Comment by WozniakMac** [comment]
@zoltanszogyenyi Thanks for review!

1. `data-drawer-toggle` stays but will work exactly this same as `data-drawer-show` and `data-drawer-hide`. It was working also without `data-modal-target` what was inconsistent. I did it just to simplify things. From now all modals will need to have `data-modal-target` because it will be tag that initialise modal.
2. The reason for local instance manager is to make sure it adds hide show event listeners only to those modals that was initialise in current `initModals` run. If you run `initModals` again it will it will create modals and add event listeners only to new html tags that were added to the page. If you have better better ideas how to make sure we don't create modal objects and event listeners on every run I'll be happy to change it. For event listeners we always check if it exists in Instance Manager. That was the issue because it Old and New modals will be there.
3. It shouldn't impact vanilla because if you run it once it works as before. Other frameworks may benefit from this if they update html on fly.


Basically I'm trying to solve 3 things here:
1. Make sure we initialise modals and event listeners only once if you run `initModals` multiple times.
2. Make sure we initialise modals and event listeners for added html if you run `initModals` again.
3. Reset all instances on `turbo:load` because it swaps the whole body with new one.

I can revert changes to `data-drawer-toggle` if you want to maintain backwards compatibility with 1.6
